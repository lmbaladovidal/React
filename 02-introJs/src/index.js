import heroes from './data/heroes'
console.log(heroes);

const getHeroesById = (id)=>{
    return heroes.find((heroe) => heroe.id === id);
}

const getHeroesByOwner = (owner)=>{
    return heroes.filter((heroe) => heroe.owner === owner);
}

console.log(getHeroesById(2));
console.log(getHeroesByOwner('DC'));
let arr = [0,1,2,3];
arr[10] = 10;
console.log(arr);
console.log(arr[9]);