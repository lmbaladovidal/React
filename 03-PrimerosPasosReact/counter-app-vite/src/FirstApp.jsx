import PropTypes from 'prop-types'
const newMessage = 'Leandro';

function getObj(){
    return {Nombre:"Leandro",
            Apellido:"Balado Vidal"}
}

export const FirstApp = ({title, subtitle,name}) => {
  console.log(title)
  return (
    <>
        <h1 data-testid="test-title">{title}</h1>
        {/* <code>{JSON.stringify(getObj())}</code> */}
        <p>{subtitle}</p>
        <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
    title : PropTypes.string.isRequired,
    subtitle : PropTypes.string,
    counter : PropTypes.number
};

FirstApp.defaultProps = {
    name :"Leandro Martin",
    //title : "No Hay titulo",
    subtitle : "No Hay subtitulo"
}
