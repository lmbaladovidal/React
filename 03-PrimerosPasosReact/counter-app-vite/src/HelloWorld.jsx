export const HelloWorld = () => {
  return (
    <h2>HelloWorld</h2>
  )
}
