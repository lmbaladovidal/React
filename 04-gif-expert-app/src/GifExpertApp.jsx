import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([ 'Kimetsu no yaiba','My Hero Academy' ]);
  const onAddCategory = (newCategory)=>{
    setCategories([newCategory,...categories]);
  }
  return (
    <>
        {/* titulo */}
        <h1>GifExpertApp</h1>
        {/* input */}
        <AddCategory 
        /* onAddCategory={setCategories} */
        onNewCategory={onAddCategory}
        />
        {/* listado de gif */}
        <ol>          
          {categories.map((category,i)=>{
            return <li key={i}>{category}</li>
          })}
        </ol>
        {/* gif items */}
    </>
  )
}
