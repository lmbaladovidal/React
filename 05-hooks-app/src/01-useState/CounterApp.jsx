import { useState } from "react"

export const CounterApp = () => {
    const [state,setCounter] = useState({
        counter0:0,
        counter1:10,
        counter2:20,
        counter3:30
    })

    const aumentarContador = ()=>{
       const auxCounter= {
            ...state,
            counter1:state.counter1+1
        }
        setCounter(auxCounter)
    }
  return (
    <>
    <h1>CounterApp</h1>
    <h4>Counter: {state.counter1}</h4>
    <h4>Counter: {state.counter2}</h4>
    <h4>Counter: {state.counter3}</h4>
    <h4>Counter: {state.counter0}</h4>
    <button className="btn" onClick={ aumentarContador }>+1</button>
    <hr />
    </>
  )
}
