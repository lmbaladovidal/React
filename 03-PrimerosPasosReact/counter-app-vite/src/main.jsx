import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import { FirstApp } from './FirstApp';
import {HelloWorld} from './HelloWorld';
import {CounterApp} from './CounterApp';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorld />
        <CounterApp value={22}/>
        {/* <FirstApp title={"Hola, soy Goku!"} subtitle='Soy un sexy subtitulo' counter={1}/> */}
    </React.StrictMode>
);