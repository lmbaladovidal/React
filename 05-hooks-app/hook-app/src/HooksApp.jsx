import React from 'react'

import { CounterApp } from './01-useState/CounterApp'
import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
import { FormWithCustomHook } from './02-useEfect/FormWithCustomHook'
import { SimpleForm } from './02-useEfect/SimpleForm'
import { FocusScreen } from './04-useRef/FocusScreen'

export const HooksApp = () => {
  return (
    <>
        <h1>HookApp</h1>
        <hr/>
{/*         <CounterApp/>
        <CounterWithCustomHook/>
        <SimpleForm/>
        <FormWithCustomHook/> */}
        <FocusScreen/>
    </>
  )
}
