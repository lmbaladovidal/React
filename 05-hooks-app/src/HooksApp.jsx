import React from 'react'

/* import { CounterApp } from './01-useState/CounterApp'
import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
import { FormWithCustomHook } from './02-useEfect/FormWithCustomHook'
import { SimpleForm } from './02-useEfect/SimpleForm'
import { MultipleCustomHooks } from './03-Examples/MultipleCustomHooks'
import { FocusScreen } from './04-useRef/FocusScreen'
import { Layout } from './05-useLayoutEffect/LayoutEffect'
import { MemoHook } from './06-Memos/MemoHook'
import { Memorice } from './06-Memos/Memorice' 
import { CallBackHook } from './06-Memos/CallBackHook'
import { Padre } from './07-tarea-memo/Padre'
import './08-useReducer/intro-reducer'*/
import { TodoApp } from './08-useReducer/TodoApp'

export const HooksApp = () => {
  return (
    <>
        <h1>HookApp</h1>
        <hr/>
{/*         <CounterApp/>
        <CounterWithCustomHook/>
        <SimpleForm/>
        <FormWithCustomHook/>
        <MultipleCustomHooks/>
        <FocusScreen/>
        <Layout/>
        <Memorice/>
        <MemoHook/> 
        <CallBackHook/>
        <Padre/>*/}
        <TodoApp/>
    </>
  )
}
