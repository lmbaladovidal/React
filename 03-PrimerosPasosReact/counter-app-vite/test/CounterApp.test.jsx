import {fireEvent, render, screen} from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"
describe('pruebas en <CounterApp/>', () => { 
    const value = 100;
    test('debe matchear con el snapshot', () => { 
        const {container} = render(<CounterApp value={value}/>);
        expect(container).toMatchSnapshot();
     });

     test('debe mostrar el valor inicial de 100', () => { 
        render(<CounterApp value={value}/>)
        expect(parseInt(screen.getByTestId("test-counter").innerHTML.trim())).toBe(100)
      });
      test('debe incrementar con el boton +1', () => { 
        render(<CounterApp value={value}/>)
        fireEvent.click( screen.getByText("+1") )  
        expect(screen.getByText('101')).toBeTruthy();
       })
      test('debe decrementar con el boton +1', () => { 
        render(<CounterApp value={value}/>)
        fireEvent.click( screen.getByText("-1") )  
        expect(screen.getByText('99')).toBeTruthy();
       })
      test('debe resetear el contador', () => { 
        render(<CounterApp value={value}/>)
        fireEvent.click( screen.getByRole('button',{name:'btn-reset'}) );  
        expect(screen.getByText('100')).toBeTruthy();
       })
 })