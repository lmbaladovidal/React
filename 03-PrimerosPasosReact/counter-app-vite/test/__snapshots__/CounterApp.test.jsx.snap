// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <CounterApp/> debe matchear con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2
    data-testid="test-counter"
  >
     
    100
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
